src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/ThrowHelper.cs

```diff
@@ -1,26 +1,97 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Microsoft.Quic;
+using System.Security.Authentication;
+using static Microsoft.Quic.MsQuic;
+
 namespace System.Net.Quic.Implementations.MsQuic
 {
     internal static class ThrowHelper
     {
-        internal static Exception GetConnectionAbortedException(long errorCode)
+        internal static QuicException GetConnectionAbortedException(long errorCode)
         {
             return errorCode switch
             {
-                -1 => new QuicOperationAbortedException(), // Shutdown initiated by us.
-                long err => new QuicConnectionAbortedException(err) // Shutdown initiated by peer.
+                -1 => GetOperationAbortedException(), // Shutdown initiated by us.
+                long err => new QuicException(QuicError.ConnectionAborted, SR.Format(SR.net_quic_connectionaborted, err), err, null) // Shutdown initiated by peer.
             };
         }
 
-        internal static Exception GetStreamAbortedException(long errorCode)
+        internal static QuicException GetStreamAbortedException(long errorCode)
         {
             return errorCode switch
             {
-                -1 => new QuicOperationAbortedException(), // Shutdown initiated by us.
-                long err => new QuicStreamAbortedException(err) // Shutdown initiated by peer.
+                -1 => GetOperationAbortedException(), // Shutdown initiated by us.
+                long err => new QuicException(QuicError.StreamAborted, SR.Format(SR.net_quic_streamaborted, err), err, null) // Shutdown initiated by peer.
             };
         }
+
+        internal static QuicException GetOperationAbortedException(string? message = null)
+        {
+            return new QuicException(QuicError.OperationAborted, message ?? SR.net_quic_operationaborted, null, null);
+        }
+
+        internal static Exception GetExceptionForMsQuicStatus(int status, string? message = null)
```

Comment:
This might read better as a switch expression.

Reply:
Unfortunately, `QUIC_STATUS_*` are not compile-time constants. Just get-only properties which hopefully get inlined+jitted as constants